projectq/meta/_loop_test.py

```diff
@@ -41,14 +41,14 @@ def test_loop_tag():
 
 def test_loop_wrong_input_type():
     eng = MainEngine(backend=DummyEngine(), engine_list=[])
-    qubit = eng.allocate_qubit()
+    eng.allocate_qubit()
```

Comment:
Sometimes we just want to create a qubit to send an allocate qubit gate and we do not care when it gets deallocated. In some cases, it is important to better control the lifetime of the qubit which is why it is sometimes required to create a variable but not in all cases.

Reply:
How is the lifetime controlled here?
If it was C++, I'd understand, but it is unclear to me in python with its optimizer and garbage collector.
Is python required to keep unused variables/references alive (at least) until the end of the scope?

Reply:
No, as far as I know the deallocation can happen at any point after the last reference to the variable dies, but I am no expert in this here.

In the two examples we have here, we could in principle not allocate any qubits since we are only concerned with testing if the context manager properly rejects some of the inputs. I have pushed a commit to remove those allocations as they are not needed.